core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -52,12 +54,14 @@ void HIP::impl_initialize(InitializationSettings const& settings) {
   KOKKOS_IMPL_HIP_SAFE_CALL(hipSetDevice(hip_device_id));
 
   // Check that we are running on the expected architecture
-  if (std::string arch_name = Impl::HIPInternal::m_deviceProp.gcnArchName;
-      arch_name.find(KOKKOS_ARCH_AMD_GPU) != 0) {
-    std::string error_message =
-        "Kokkos::HIP::initialize ERROR: running kernels compiled for " +
-        std::string(KOKKOS_ARCH_AMD_GPU) + " on " + arch_name + " device.\n";
-    Kokkos::abort(error_message.c_str());
+  if (Kokkos::show_warnings()) {
```

Comment:
You might want to add a comment explaining why this is a warning rather than an error